src/Common/IntervalKind.cpp

```diff
@@ -9,13 +9,13 @@ namespace ErrorCodes
     extern const int SYNTAX_ERROR;
 }
 
-Int32 IntervalKind::toAvgSeconds() const
+Float64 IntervalKind::toAvgSeconds() const
```

Comment:
I think Float64 may introduce some unnecessary rounding errors for callers. Is that okay (yes if use is only e.g. for prettyprinting)? In the end, we deal with an integral (time) amount of a base unit.

One option would be to change the unit of this method from seconds to nanoseconds, i.e.
```
  case IntervalKind::Nanosecond: return 1;
  case IntervalKind::Microsecond: return 1000;
  ...
```
+ to rename it toAvcNanoSeconds(). I actually think the return type needs to be changed from Int32 --> Int64 to be able to represent a year in nanoseconds. Because of the large returned values and because nanoseconds seems a *very* small unit for an interval, this option is not very appealing, admittedly.

So one question is if micro and nano seconds can ever occur in practice (e.g. in SQL)? If no, we might only support milliseconds, seconds, minutes etc. here and make milliseconds the base unit.

In Parsers/Access/ASTCreateQuotaQuery.cpp, we divide by the value returned by toAvgSeconds(). Guess this is why you no longer return 0. So another option would be to retain the existing toAvgSeconds() + to document it's behavior (may return 0) and to adjust the caller to check for 0.

Reply:
I guess returning 0 in such a function was not (my own) best idea, because generally we wouldn't expect it to be a zero value. I think it is completely OK there